sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java

```diff
@@ -213,16 +216,46 @@ public void error(SchedulerDriver driver, String message) {
         SchedulerUtils.hardExit(SchedulerErrorCode.ERROR);
     }
 
+    protected void suppressOrRevive(PlanCoordinator planCoordinator) {
+        synchronized (suppressReviveLock) {
```

Comment:
Question: I've seen `synchronized (this)` this used in some places online (e.g. http://winterbe.com/posts/2015/04/30/java8-concurrency-tutorial-synchronized-locks-examples/). Is there an advantage over using an explicit object other than readability (and the fact that one can only handle one lock with `synchronized (this)`?